Добавлены библиотеки:

psasha007 opened the pull request:
maven.compiler.source 11
allure.version 2.19.0
org.mockito 4.8.0
com.google.code.gson 2.10
io.rest-assured 5.3.0
org.projectlombok 1.18.24
org.seleniumhq.selenium 4.6.0

Добавлен файл .gitignore

psasha007 committed:
Добавлены библиотеки:
maven.compiler.source 11
allure.version 2.19.0
org.mockito 4.8.0
com.google.code.gson 2.10
io.rest-assured 5.3.0
org.projectlombok 1.18.24
org.seleniumhq.selenium 4.6.0

Добавлен файл .gitignore

psasha007 committed:
Обозначил классы
BunTest
BurgerTest
IngredientTest
IngredientTypeTest

psasha007 committed:
Оставил в pom библиотеки: Jacoco, Mockito, JUnit 4.

psasha007 committed:
Добавил тесты для проверки методов класса Bun
equalGetNameBunTest - Проверка метода названия Булочки
equalGetPriceBunTest - Проверка метода назначенной цены Булочки

Попробовал добавить jacoco

psasha007 committed:
Добавил тесты для проверки методов класса Burger
Для проверок использовал параметризацию и Mock
equalBunBurgerTest - Проверка метода сборки бургера
equalIngredientBurgerTest - Проверка метода добавления ингредиента
equalMoveIngredientTest - Проверка метода перемещения слоя с ингредиентом
equalRemoveIngredientTest - Проверка метода удаления ингредиентов

psasha007 committed:
Добавил тесты для проверки методов класса IngredientTest
equalGetNameIngredientTest - Проверка метода названия ингредиента
equalGetPriceIngredientTest - Проверка метода назначенной цены ингредиента
equalGetTypeIngredientTest - Проверка метода типа ингредиента

psasha007 committed:
Добавил проверки Jacoco

psasha007 committed:
мелкие правки

nikolay-977 commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,173 @@
+        return new Object[][]
+                {// передали тестовые данные
+                        {new Bun("black bun", 100),
+                                new Ingredient(IngredientType.SAUCE, "hot sauce", 100),
```
Необходимо исправить: Тесты для класса Burger должны быть изолированы от классов Bun и Ingredient. При написании тестов для класса Burger во всех кейсах используй моки и стабы классов Bun и Ingredient, а не реальные объекты.

psasha007 replied:
Исправил тесты для  класса Burger

nikolay-977 commented on src/test/java/BurgerTest.java:
```diff
@@ -0,0 +1,173 @@
+        burger.setBuns(bunsList.get(0));
+        String expected = bunPrint + bunPrint + "\r\n" + totalPriceBun;
+
+//        System.out.println(expected);
```
Необходимо исправить: удали лишний комментированный код

psasha007 replied:
удалил лишние комментарии

nikolay-977 commented on src/test/java/IngredientTest.java:
```diff
@@ -0,0 +1,47 @@
+    public void equalGetNameIngredientTest() throws Exception {
+        String expected = "NameIngredient";
+        String actual = ingredient.getName();
+        Assert.assertEquals("Неверное название ингредиента",expected, actual);
```
Необходимо исправить: Используй сочетание клавиш Ctrl+Alt+L (Option+Command+L для Mac) для расстановки пробелов и табуляции в соответствии с настройками Code Style в Idea. (Или в VSC: Shift + Alt + F
и Shift + Option + F для Mac)

psasha007 replied:
спасибо, классная штука

nikolay-977 commented on src/test/java/IngredientTypeTest.java:
```diff
@@ -0,0 +1,4 @@
+public class IngredientTypeTest {
+    // в классе IngredientType нет методов, непонятно нужно здесь что-то проверять
```
Необходимо исправить: Напиши тесты для класса IngredientType в классе IngredientTypeTest. Например, можно использовать метод valueOf().

psasha007 replied:
Написал тесты для класса IngredientType

psasha007 committed:
- Добавил тесты для класса IngredientType

psasha007 committed:
- Добавил тесты для класса IngredientType

psasha007 committed:
- Поправил тесты для класса Burger, изолировал от классов Bun и Ingredient
- Удалил лишний комментированный код
- Спасибо, теперь использую Ctrl+Alt+L для расстановки пробелов и табуляции
- Добавил тесты для класса IngredientType
